New RSTUF Guide (#341)
repository-service-tuf/repository-service-tuf: docs/source/guide/general/keys.rst

############

> Singing Keys

Reviewer: Are these "signing keys" ?

############

Repository Service for TUF (RSTUF) requires two sets of keys for
:ref:`guide/deployment/index:Deployment` and
:ref:`guide/deployment/setup:Service Setup`:
:ref:`guide/general/keys:Root Key(s) (offline)` and the
:ref:`guide/general/keys:Online Key`.

Root Key(s) (offline)
#####################

> The Root key(s) delegates trust to TUF. The number of keys is
> the number of identities/people who administrate the top-level TUF Metadata.

Reviewer: ```suggestion
The Root key(s) delegates trust to TUF. The number of keys is
the number of identities/people who administer the top-level TUF Metadata.
```

RSTUF requires all Root key(s) only during the
:ref:`guide/deployment/setup:Service Setup` specifically during the
:ref:`guide/deployment/setup:Ceremony` process. This process also defines the
Root key threshold, representing the number of Root key(s) for future offline
operations.

.. note::
  .. collapse:: See the number of Root keys/threshold example

>       An organization defines that it will have 5 (five) Root keys to
>       administrate the RSTUF Service. A total of 5 (five) people will require
>       to use their keys individually during the Ceremony process.

Reviewer: ```suggestion
      An organization declares that it will utilize 5 (five) Root keys in order to
      administer the RSTUF Service. All 5 (five) people will be required
      to utilize their keys individually during the Ceremony process.
```

Reviewer: So, I'm not sure if all five people will need to participate in the Ceremony, or they simply have the option of participating. Perhaps you need to explain this to me.

Author: All five people will need to participate in the ceremony once they all need to load their keys.
We are working on a feature where they can participate in this ceremony asynchronously.

      During the Ceremony process, the same organization defines that the
      threshold for Root metadata is 2 (two).

      .. code::

        Root keys: 5
        Root keys threshold: 2

      The organization to perform :ref:`guide/general/usage:Metadata Update`
      process requires at least 2 (two) people to use their keys.

.. caution::
  The root key(s)
  `should be stored secured offline <https://theupdateframework.github.io/specification/latest/#key-management-and-migration>`_.

The key must be compatible with `Secure Systems Library <https://github.com/secure-systems-lab/securesystemslib>`_.

Online Key
##########

The online key signs TUF metadata for the roles
:ref:`that use the online key<guide/general/Introduction:TUF Metadata>`.

RSTUF requires the online key during the
:ref:`guide/deployment/setup:Service Setup`, specifically during the
:ref:`guide/deployment/setup:Ceremony` process.

During RSTUF Worker service deployment, configure the online key using a
supported `Key Vault Service <https://repository-service-tuf.readthedocs.io/en/latest/guide/repository-service-tuf-worker/Docker_README.html#required-rstuf-keyvault-backend>`_

.. caution::

>   * Do not expose the online private key.

Reviewer: "private" or "privacy" ?

Author: On the key context, we have[ two definitions for it](https://en.wikipedia.org/wiki/Digital_signature): private key and public key

Reviewer: If it's the specific name of a tool, the "private key" then it stands. If it reference the general privacy of the key, then the grammatical phrase would be "privacy key".
